### ISSUE TITLE
installing DirectX on the VM

### ISSUE THREAD

**Issue by luky92** [jamesstringerparsec/Easy-GPU-P#54]
I have a game that requires specific version of DirectX installed is there a way to install it on such VM?

**Comment by Thijskunst** [comment]
is it possible that you're using a geforce 10-series gpu?

**Comment by luky92** [comment]
Yes GTX 1060 6gb

**Comment by Mrkh97** [comment]
I’m having the same problem with a 3070

**Comment by Thijskunst** [comment]
im having this issue with a 1080 and 1070ti. the errors in dxdiag tell me its something with the audio part of directX

**Comment by luky92** [comment]
I get exactly the same error

**Comment by luky92** [comment]
@jamesstringerparsec do you have any idea what it might have to do with?

**Comment by Gravlok** [comment]
Audio part from DXdiag is not relevant. DirectX from dxwebsetup will fail to install (reason is unknown to me). I fixed it with DX Dependency Fixer:
https://obsproject.com/forum/resources/directx-dependency-fixer-dxfix.87/

**Comment by Thijskunst** [comment]
I got it working!!!! the issue seems to be the nvidia driver. 

I downgraded my driver from 511.something to 510.06.
 then i mounted the VM drive and copied the driver files from "C:\Windows\System32\DriverStore\FileRepository\nv_dispi.inf_amd64_xxxxxxxxxx" on the host
 to  
"H:\Windows\System32\HostDriverStore\FileRepository\" on the VM drive (drive H in my case)
after that i enabled the "microsoft hyper-v video" adapter in thew VM device manager

**Comment by reno911unlocred** [comment]
> I got it working!!!! the issue seems to be the nvidia driver.
> 
> I downgraded my driver from 511.something to 510.06. then i mounted the VM drive and copied the driver files from "C:\Windows\System32\DriverStore\FileRepository\nv_dispi.inf_amd64_xxxxxxxxxx" on the host to "H:\Windows\System32\HostDriverStore\FileRepository" on the VM drive (drive H in my case) after that i enabled the "microsoft hyper-v video" adapter in thew VM device manager

Have you tried with any other versions of the driver?

**Comment by Thijskunst** [comment]
> Have you tried with any other versions of the driver?

Yes, but  510.06 is the only one that worked. all the other drivers gave me the same problems. 
download link for the driver : 
https://developer.nvidia.com/51006-gameready-win11-win10-dch-64bit-international
(source: https://forums.developer.nvidia.com/t/new-cuda-on-wsl2-driver-510-06-is-now-live/188333)

**Comment by reno911unlocred** [comment]
Does this gave you a kernel error when trying to do so with a different driver?

**Comment by Thijskunst** [comment]
update 2: 
i got it working perfectly without  any directX issues. all games seem to be working.  it looks like it was a windows issue inside the VM. the script installs windows home/pro N. i think that is part of the issue.  
how to:
1. create vm using script (VM1)
2. make 2nd vm and install directX and nvidia/amd gpu driver (VM2)
3. move the HDD file from VM 2 to the folder of VM1
4. rename the file from VM2 to the name of the file of VM1 and give the file from VM1 a different name
5. boot VM1 and it should work

driver location on VM:  C:\Windows\System32\HostDriverStore\FileRepository.

**Comment by reno911unlocred** [comment]
> update 2: i got it working perfectly without any directX issues. all games seem to be working. it looks like it was a windows issue inside the VM. the script installs windows home/pro N. i think that is part of the issue. how to:
> 
> 1. create vm using script (VM1)
> 2. make 2nd vm and install directX and nvidia/amd gpu driver (VM2)
> 3. move the HDD file from VM 2 to the folder of VM1
> 4. rename the file from VM2 to the name of the file of VM1 and give the file from VM1 a different name
> 5. boot VM1 and it should work
> 
> driver location on VM: C:\Windows\System32\HostDriverStore\FileRepository.

Bur bro, the script installs whatever windows iso you tell it to, maybe you are ising the wrong iso or the one that the windows media creation tool downloads for you, may I ask, where did you got the iso from?

**Comment by doublep21** [comment]
> > update 2: i got it working perfectly without any directX issues. all games seem to be working. it looks like it was a windows issue inside the VM. the script installs windows home/pro N. i think that is part of the issue. how to:
> > 
> > 1. create vm using script (VM1)
> > 2. make 2nd vm and install directX and nvidia/amd gpu driver (VM2)
> > 3. move the HDD file from VM 2 to the folder of VM1
> > 4. rename the file from VM2 to the name of the file of VM1 and give the file from VM1 a different name
> > 5. boot VM1 and it should work
> > 
> > driver location on VM: C:\Windows\System32\HostDriverStore\FileRepository.
> 
> But bro, the script installs whatever windows iso you tell it to, maybe you are using the wrong iso or the one that the windows media creation tool downloads for you, may I ask, where did you got the iso from?
> 
> Edit: typos.

same here , windows 11 N . I donwloaded directly from microsoft

**Comment by doublep21** [comment]
> update 2: i got it working perfectly without any directX issues. all games seem to be working. it looks like it was a windows issue inside the VM. the script installs windows home/pro N. i think that is part of the issue. how to:
> 
> 1. create vm using script (VM1)
> 2. make 2nd vm and install directX and nvidia/amd gpu driver (VM2)
> 3. move the HDD file from VM 2 to the folder of VM1
> 4. rename the file from VM2 to the name of the file of VM1 and give the file from VM1 a different name
> 5. boot VM1 and it should work
> 
> driver location on VM: C:\Windows\System32\HostDriverStore\FileRepository.

just modify Edition    = 6 to--> Edition    = 5 in the script
